src/pricing.ts

```diff
@@ -476,6 +477,46 @@ const recomputeDetailTotalsFromCompositePrice = (
   }, details || initialPricingDetails);
 };
 
+const BLACK_LISTED_KEYS: Set<keyof Price> = new Set([
+  '_org',
+  '_schema',
+  '_created_at',
+  '_updated_at',
+  '_owners',
+  '_acl',
+  '_acl_sync',
+  '_viewers',
+  '_relations',
+  '$relation',
+  'file',
+  '_files',
+  'workflows',
+  '_slug',
+]);
+
+/**
+ * Maps the whitelisted keys from the _price that we want available, avoiding having unnecessary data in the metadata.
+ *
+ *? @todo We should define _price as not being an optional key in PriceItemDto since there is no case where is optional
+ *? Additionally, we should also define the _product as a separate interface to avoid `PriceItemDto['_product']` type definitions.
+ */
+export const getMappedPricing = (price: Price): Price => {
```

Comment:
Can we make it more specific? This is a mapper to be used when we want to snapshot a price only right?

```suggestion
export const mapToPriceSnapshot = (price: Price): Price => {
```

Reply:
Well the method is generic as it is, living it up to interpretation of "MappedPricing" means, to me it doesn't mean much. Also if we leave it with a generic name, then it suggests this "MappedPricing" is open to change for other purposes and it's not. This function is a specific mapper to give us the Price Snapshot data we will store within line_items and item_components. And it's naming should suggest that.

Reply:
Alright, fair enough! I'll have a look today when I finish some other things up with this PR and change the name. Thank you 🙂